### DISCUSSION TITLE
brew list with "License" field - do I have non-free software installed?

### DISCUSSION THREAD

**Discussion by GETandSELECT** [Homebrew/discussions#10] [Everyday usage] [answered]
I wish to see an overview of all formula I installed and their license. `brew list` with license tag.

Goal is to see if I installed proprietary / non-free software.

```
brew info prometheus
prometheus: stable 2.21.0 (bottled)
Service monitoring system and time series database
https://prometheus.io/
Not installed
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/prometheus.rb
License: Apache-2.0
```

**Comment by gromgit** [comment, marked as answer]
The first paragraph of https://docs.brew.sh/License-Guidelines reads:
> We only accept formulae that use a Debian Free Software Guidelines license or are released into the public domain following DFSG Guidelines on Public Domain software.

So if you only use the core tap, your result set is guaranteed to be empty.

If you use third-party taps, there's no guarantee that they have `license` stanzas, but a fairly simple way to get the info you want involves `jq`:
```
brew info --json --installed | jq -r '.[] | [.name, .license] | @tsv' | while IFS=$'\t' read name license; do
  # Check $license for non-free indicators
  ...
done
```

**Comment by brettp** [comment]
This doesn't work because there's no verification or requirements for the license field. 

Multiple known OSS in the default cask repo don't have a license field (see [Blender](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/b/blender.rb), [Gimp](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/g/gimp.rb), [Audacity](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/a/audacity.rb)).

Commercial software also fails this test for the same reason. See the recently added [Limitless](https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/l/limitless.rb) commercial and proprietary SaaS AI service as an example. 

There's unfortunately no way to check if you're using only OSS software in Homebrew without checking each project's individual license from their own source or website (since commercial and non-licensed software is allowed in casks).

**Reply by SMillerDev** [reply]
Casks don't have a license since there is no requirement for them to have one to be included. In addition to that, they are often proprietary and hard to represent in a simple statement.

**Reply by SMillerDev** [reply]
> There's unfortunately no way to check if you're using only OSS software in Homebrew

There is a simple one: do you install software outside of Homebrew/core? You then you probably have non-OSS software installed

**Comment by brettp** [comment]
Thank you for the confirmation. My point is the selected answer doesn't work because the license field is not required and in addition to proprietary, many open source projects don't even use it. 

The field is essentially unreliable and not an accurate representation, so either wouldn't be an answer to OP's question, _or_ should be enforced to be present and accurate.

**Reply by SMillerDev** [reply]
That's not accurate, OP asked about Formula. Within the Homebrew organisation those only live in Homebrew/core and although they are not enforced for existing Formula, the vast majority of Formula does have it defined.

You brought in casks, but that is not the original question.

**Reply by brettp** [reply]
Thanks. I do understand there's a technical difference. From a UX perspective, and to the core question OP is asking, it feels like a difference without distinction though. OP's title `brew list with "License" field - do I have non-free software installed?` doesn't single out formulae, and the tool presents information nearly identically for both casks and formulae. 

Just pointing this out because for folks / projects that must confirm licenses, the solution here doesn't work with the most common configurations of the tool.

**Reply by GETandSELECT** [reply]
After this question, I learned that there are no problematic licenses in core. However, there are a lot of proprietary casks, so I would really like to see a separation between FLOSS and proprietary casks. I would prefer not to use proprietary apps on my desktop. That would be my feature request, have there already been discussions along similar lines?

**Reply by p-linnane** [reply]
It would be easiest for you to assume that casks are proprietary and not open source. There are some exceptions to this, but given the overwhelming majority of casks are proprietary, it's a safe bet.
